### ISSUE TITLE
passLP resets the objective sense

### ISSUE THREAD

**Issue by lovasoa** [ERGO-Code/HiGHS#484]
Hello,
  I recently spent some time on an issue before realizing that when building a problem, `Highs_passLp` has to be called BEFORE `Highs_changeObjectiveSense`. I think this should either be documented (both in the documentation for passLp and changeObjectiveSense, or the objective sense should not be lost after passLp.

**Comment by odow** [comment]
I guess this and #480 are really just "Document the C/C++ API" :smile:

**Comment by jajhall** [comment]
Indeed, more of a philosophical issue. My take on it is that when HiGHS is empty there's no objective whose sense can be changed - in which case Highs_changeObjectiveSense should return a warning, I admit. Since passLp can't take "sense" (or "offset"), the LP is assumed to be a minimization (with zero offset). 
For reasons of clarity, I think that the solution is to add sense" and "offset" to passLp. I'm also realising that the C API needs to offer more. I'd anticipated that people would work more with the C++ API.
At the risk of confusing less experienced users, do you feel that I should add "orientation" to passLp so that users can pass the matrix row-wise or column-wise? Rather than just expect users to transpose the matrix themselves!

**Comment by lovasoa** [comment]
I don't really have a problem with the current behavior, other than that it is not documented and does not return an error when used incorrectly. I don't think it's worth breaking backwards compatibility by changing the API. Adding one line in the doxygen docs of passLp and changeObjectiveSense, and one `if` in changeObjectiveSense would be a perfectly good solution IMHO.

I am not familiar with the word "offset" in this context. Does this mean equality constraints ? If so, then I don't think the API needs to be changed at all. Each row already has both a lower and an upper bound, so the "offset" is the set of constraints for which low==high.

**Comment by odow** [comment]
Off set is the objective `c'x + offset`. 

Also think no need to change the C API. It's pretty good, just needs documentation, but that can probably be done by people outside the core development team. Your time is better spent writing code :smile:.

**Comment by lovasoa** [comment]
A lot off developers tend to think that, but this often results in poor documentation, and prevents adoption of the software in the long run. Do you have enough non-developer resources able to write good technical documentation ?

**Comment by jajhall** [comment]
I certainly don't expect others to write the documentation, but HiGHS is a team of three so we have no non-developer resources to write good technical documentation. Once we've got a first formal release, I can give more time to writing documentation. Fortunately I can spend more than half of my working time on HiGHS.
The documentation on https://www.highs.dev/ is a bit out-of-date, and I'd be interested to see what you feel we need to do when I revise it. I know that the C API isn't documented, but I aimed it at users of the C++ API - maybe that's an error. The documentation on the C (and other APIs) would be to give the equivalent method names in the C++ API, allowing a functional description to be read in just one place.
